format support

Yvonne-Aizawa opened the pull request:
with ollama 0.1.9  you can send if you want the output to be formatted.
for now it only supports JSON. but when left empty it generates as normal.

I added a parameter to GenerationRequest to specify if it needs to be formatted as json. when it is None it will leave out the parameter and generate as normal. I made it an enum as i seems like the format parameter might get more types and it would be easy to add new ones to it.

pepperoni21 commented:
Hey, I don't think we need to support other formats than json since we're using Serde, the user does not matter in what format we get the data as they would just access it through the Rust bindings that this library provides.

Yvonne-Aizawa commented:
I do not know if i understand it right. But this is how ollama works now [docs](https://github.com/jmorganca/ollama/blob/main/docs/api.md#json-mode). it forces the llm to return valid json. this just got added in ollama 0.1.9. you still recieve the data as a string. but the string would be a valid json object

Yvonne-Aizawa commented:
example with formatting to json on
> what is the weather like?
{"message": "I'm just an AI, I don't have access to real-time weather information. However, I can tell you that the weather can vary depending on your location and time of year. Could you please provide me with more context or specify a location you are interested in?", "request_id": 123456}

example with formatting none
> what is the weather like?
I'm just an AI, I don't have real-time access to current weather conditions. However, I can tell you the typical weather patterns for a place if you provide me with the location. Please let me know the city or region you are interested in, and I will do my best to give you an idea of what the weather is like there.

pepperoni21 commented on README.md:
```diff
@@ -37,7 +37,7 @@ if let Ok(res) = res {
 let prompt = "Why is the sky blue?".to_string();
 
-let mut stream = ollama.generate_stream(GenerationRequest::new(model, prompt)).await.unwrap();
+let mut stream = ollama.generate_stream(GenerationRequest::new(model, prompt,None)).await.unwrap();
```
I think the format should be None by default without having to add it to the new method, like the other optional fields, as it meets a very particular need.

pepperoni21 commented on src/generation/completion/request.rs:
```diff
@@ -13,18 +13,25 @@ pub struct GenerationRequest {
             system: None,
             template: None,
             context: None,
+            //format value for now just json
```
```suggestion
            // The format to return a response in. Currently the only accepted value is `json`
```

pepperoni21 commented on tests/generation.rs:
```diff
@@ -42,7 +43,22 @@ async fn test_generation() {
         .await
         .unwrap();
 }
+#[tokio::test]
+async fn test_generation_json() {
+    let ollama = Ollama::default();
+
+    let _ = ollama
+        .generate(GenerationRequest::new(
+            "llama2:latest".to_string(),
+            PROMPT.into(),
+            Some(FormatEnum::Json)
+            
+        ))
+        .await
+        .unwrap();
+}
```
I don't think we need a test for that

Yvonne-Aizawa commented:
removed parameter from the new() method. and updated code to reflect it
altered the comment to explain better what the field does
removed the test from generation.rs
